zkchannels-contract/zkchannel_smartpy_script.py

```diff
@@ -14,193 +14,240 @@
 MERCH_PK5_G2 = "0x1304a722c780f8b4973dd4da42ef4148af2a580aa3aeddbdaba604a86ec6e62750d699bd13647089278a1e6cc490986f181529059281216c836f054f392efb90b4890a57e46f43f7dc5a8faf0fe41a1b2cd54402dd0af86b78c3a8e175daf9530a2d9d970935dc3e93463565b431d38e13456092bce8da73ed1c2274a02dd29e1e3e0dda7a6f1e0f6c67ab741b4cc20212dcab1cad18c655264f6f56a9ad1a383be2cd0c72d2fdb59ffea76cb1c9d57f84a0d82ea391579bb5e11bc61e40d136"
 CLOSE_FLAG_B = "0x000000000000000000000000000000000000000000000000000000434c4f5345"
- 
-AWAITING_FUNDING = 0
-OPEN = 1
-EXPIRY = 2
-CUST_CLOSE = 3
-CLOSED = 4
- 
-ZERO_IN_G1 = "0x400000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000"
+
+# zkChannel contract statuses
+AWAITING_CUST_FUNDING = 0
+AWAITING_MERCH_FUNDING = 1
+OPEN = 2
+EXPIRY = 3
+CUST_CLOSE = 4
+CLOSED = 5
+FUNDING_RECLAIMED = 6
+
+ # This is the value of the identity element in the elliptic curve pairing group G1 from BLS12-381 
+ # (i.e., the point at infinity on BLS12-381).
+IDENTITY_IN_G1 = "0x400000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000"
  
 
-# cid is a unique identifier for the channel.
-# Addresses are used both for interacting with contract, and receiving payouts. Addresses must be for implicit accounts (tz1) only, not smart contracts.
-# Public keys are used for verifying signatures required for certain state transitions.
-# revLock is the revocation lock used to punish a customer who broadcasts a revoked custState.
-# selfDelay defines the delay (in seconds) during which the other party can counter specific state transitions.
-# delayExpiry is the unix timestamp corresponding to when the delay expires.
 class ZkChannel(sp.Contract):
 
+    # Returns true if this element is the identity element (the point at infinity).
     @sp.global_lambda
-    def is_g1_not_zero(val):
+    def is_g1_identity(val):
         packed_s1 = sp.pack(val)
-        packed_zero = sp.to_constant(sp.pack(sp.bls12_381_g1(ZERO_IN_G1)))
-        sp.result(packed_s1 != packed_zero)
+        packed_zero = sp.to_constant(sp.pack(sp.bls12_381_g1(IDENTITY_IN_G1)))
+        sp.result(packed_s1 == packed_zero)
 
-    def __init__(self, cid, custAddr, merchAddr, custPk, merchPk, custFunding, merchFunding, selfDelay, g2, merchPk0, merchPk1, merchPk2, merchPk3, merchPk4, merchPk5, close_flag):
+    # __init__ initializes the contract's storage at the time of origination. All the arguments 
+    # must be provided in the origination operation. 
+    def __init__(self, cid, custAddr, merchAddr, custPk, merchPk, custFunding, merchFunding, selfDelay, g2, y2s_0, y2s_1, y2s_2, y2s_3, y2s_4, x2, close_flag):
```

Comment:
making a note at the end to update all the terms throughout the contract to match libzkchannels-crypto/zeekoe.

Reply:
In general, Rust variable names closely match their type names. Types are written in `UpperCamelCase`, while variable and function names are `snake_case` An all-caps variable name indicates a constant value. In this case, since the close flag is a parameter without a fixed value, I would call it `close_scalar`. 

`custAddr` is a field in a `CustomerFundingInfomation` struct called `address`. Here I would call it `customer_address` or maybe `customer_funding_address`. Ditto `merchant_address`. The type of these in zeekoe is `tezedge::ImplicitAddress`, which I renamed `TezosFundingAddress` because it was easier for me to remember what it meant.

I call funding amounts `merchant_balance` and `customer_balance`.

I haven't had to use a `selfDelay` yet but following Rust naming conventions, I would use `self_delay`.

Tezos public keys are also a field in the `XFundingInformation` struct called  `public_key`, with type `TezosPublicKey` AKA `tezedge::PublicKey`. Here I would probably call them `merchant_public_key` and `customer_public_key`. If you do end up wrapping up the PS key into a single thing, I would call it something like `merchant_zkabacus_public_key` to differentiate it from the Tezos ones.

I would not mind terribly if you kept the convention of `lowerCamelCase` variable names here. I would far prefer consistency in variable name style over a mix of camel and snake case names.

Do variable name lengths count toward gas/storage cost in Tezos or do they get compiled away into smaller representations?